src/constructs/ecs/ecs-task.ts

```diff
@@ -188,6 +197,12 @@ export class GuEcsTask extends Construct {
       );
     }
 
+    const taskSubnets = subnets
+      ? subnets
+      : assignPublicIp
+        ? GuVpc.subnetsFromParameterFixedNumber(scope, { type: SubnetType.PUBLIC, app }, 3)
+        : GuVpc.subnetsFromParameterFixedNumber(scope, { type: SubnetType.PRIVATE, app }, 3);
+
```

Comment:
I think this nested ternary is unnecessary complexity; could we remove the `assignPublicIp` prop and pass in the correct `subnets` when instantiating this construct? The consequence is the `subnets` prop becomes mandatory, but I think it'll result in a simpler interface. For example:

```ts
new GuEcsTask(scope, "my-task", {
  subnets: GuVpc.subnetsFromParameterFixedNumber(scope, { type: SubnetType.PUBLIC, app }
});
```

Reply:
Good idea, thanks